addons/sale/models/sale_order.py

```diff
@@ -1189,6 +1189,8 @@ def _send_order_confirmation_mail(self):
         """
         for order in self:
             mail_template = order._get_confirmation_template()
+            salesperson = self.env['website'].get_current_website().salesperson_id
+            mail_template = mail_template.with_user(salesperson)
```

Comment:
Didn't notice earlier, but this won't work.
There cannnot be `website` logic in `sale`, and you cannot rely on the current website for code that is run mostly through crons.
Can't you use the salesperson of the SO instead ?